<a href="https://colab.research.google.com/github/Manikandan-professional/LoanApprovalPrediction/blob/main/Loan_Approval_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [802]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
from sklearn.preprocessing import LabelEncoder

In [803]:
train='https://raw.githubusercontent.com/Manikandan-professional/Datasets/main/Training%20Dataset.csv'
test='https://raw.githubusercontent.com/Manikandan-professional/Datasets/main/Test%20Dataset.csv'
df_train=pd.DataFrame((pd.read_csv(train)))
df_test=pd.DataFrame((pd.read_csv(test)))
df_train.head()

Loan_ID Gender Married  ... Credit_History Property_Area Loan_Status
0  LP001002   Male      No  ...            1.0         Urban           Y
1  LP001003   Male     Yes  ...            1.0         Rural           N
2  LP001005   Male     Yes  ...            1.0         Urban           Y
3  LP001006   Male     Yes  ...            1.0         Urban           Y
4  LP001008   Male      No  ...            1.0         Urban           Y

[5 rows x 13 columns]

# **Train Dataset**

# **Data Cleaning**

In [804]:
df_train=df_train.drop(columns=['Loan_ID'])
df_train

Gender Married Dependents  ... Credit_History Property_Area  Loan_Status
0      Male      No          0  ...            1.0         Urban            Y
1      Male     Yes          1  ...            1.0         Rural            N
2      Male     Yes          0  ...            1.0         Urban            Y
3      Male     Yes          0  ...            1.0         Urban            Y
4      Male      No          0  ...            1.0         Urban            Y
..      ...     ...        ...  ...            ...           ...          ...
609  Female      No          0  ...            1.0         Rural            Y
610    Male     Yes         3+  ...            1.0         Rural            Y
611    Male     Yes          1  ...            1.0         Urban            Y
612    Male     Yes          2  ...            1.0         Urban            Y
613  Female      No          0  ...            0.0     Semiurban            N

[614 rows x 12 columns]

In [805]:
df_train.shape

(614, 12)

In [806]:
df_train.describe()

ApplicantIncome  CoapplicantIncome  ...  Loan_Amount_Term  Credit_History
count       614.000000         614.000000  ...         600.00000      564.000000
mean       5403.459283        1621.245798  ...         342.00000        0.842199
std        6109.041673        2926.248369  ...          65.12041        0.364878
min         150.000000           0.000000  ...          12.00000        0.000000
25%        2877.500000           0.000000  ...         360.00000        1.000000
50%        3812.500000        1188.500000  ...         360.00000        1.000000
75%        5795.000000        2297.250000  ...         360.00000        1.000000
max       81000.000000       41667.000000  ...         480.00000        1.000000

[8 rows x 5 columns]

In [807]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Gender             601 non-null    object 
 1   Married            611 non-null    object 
 2   Dependents         599 non-null    object 
 3   Education          614 non-null    object 
 4   Self_Employed      582 non-null    object 
 5   ApplicantIncome    614 non-null    int64  
 6   CoapplicantIncome  614 non-null    float64
 7   LoanAmount         592 non-null    float64
 8   Loan_Amount_Term   600 non-null    float64
 9   Credit_History     564 non-null    float64
 10  Property_Area      614 non-null    object 
 11  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(7)
memory usage: 57.7+ KB


In [808]:
df_train['Married'].value_counts()

Yes    398
No     213
Name: Married, dtype: int64

In [809]:
#df_train=pd.get_dummies(df_train)

In [810]:
df_train.head()

Gender Married Dependents  ... Credit_History Property_Area  Loan_Status
0   Male      No          0  ...            1.0         Urban            Y
1   Male     Yes          1  ...            1.0         Rural            N
2   Male     Yes          0  ...            1.0         Urban            Y
3   Male     Yes          0  ...            1.0         Urban            Y
4   Male      No          0  ...            1.0         Urban            Y

[5 rows x 12 columns]

In [811]:
df_train=df_train.dropna()

In [812]:
df_train=df_train.reset_index()

In [813]:
df_train

index  Gender Married  ... Credit_History Property_Area Loan_Status
0        1    Male     Yes  ...            1.0         Rural           N
1        2    Male     Yes  ...            1.0         Urban           Y
2        3    Male     Yes  ...            1.0         Urban           Y
3        4    Male      No  ...            1.0         Urban           Y
4        5    Male     Yes  ...            1.0         Urban           Y
..     ...     ...     ...  ...            ...           ...         ...
475    609  Female      No  ...            1.0         Rural           Y
476    610    Male     Yes  ...            1.0         Rural           Y
477    611    Male     Yes  ...            1.0         Urban           Y
478    612    Male     Yes  ...            1.0         Urban           Y
479    613  Female      No  ...            0.0     Semiurban           N

[480 rows x 13 columns]

In [814]:
df_train=df_train.iloc[:,1:]

In [815]:
df_train

Gender Married Dependents  ... Credit_History Property_Area  Loan_Status
0      Male     Yes          1  ...            1.0         Rural            N
1      Male     Yes          0  ...            1.0         Urban            Y
2      Male     Yes          0  ...            1.0         Urban            Y
3      Male      No          0  ...            1.0         Urban            Y
4      Male     Yes          2  ...            1.0         Urban            Y
..      ...     ...        ...  ...            ...           ...          ...
475  Female      No          0  ...            1.0         Rural            Y
476    Male     Yes         3+  ...            1.0         Rural            Y
477    Male     Yes          1  ...            1.0         Urban            Y
478    Male     Yes          2  ...            1.0         Urban            Y
479  Female      No          0  ...            0.0     Semiurban            N

[480 rows x 12 columns]

In [816]:
le=LabelEncoder()

# **Gender Analysis**

In [817]:
print('The number of unique values in the gender column is:\n',df_train['Gender'].value_counts())

The number of unique values in the gender column is:
 Male      394
Female     86
Name: Gender, dtype: int64


In [818]:
df_train['Gender']=le.fit_transform(df_train['Gender'])

In [819]:
df_train.head()

Gender Married Dependents  ... Credit_History Property_Area  Loan_Status
0       1     Yes          1  ...            1.0         Rural            N
1       1     Yes          0  ...            1.0         Urban            Y
2       1     Yes          0  ...            1.0         Urban            Y
3       1      No          0  ...            1.0         Urban            Y
4       1     Yes          2  ...            1.0         Urban            Y

[5 rows x 12 columns]

In [820]:
df_train['Gender']=df_train['Gender'].astype(int)

# **Married Analysis**

In [821]:
print('The number of unique values in the Married column is:\n',df_train['Gender'].value_counts())

The number of unique values in the Married column is:
 1    394
0     86
Name: Gender, dtype: int64


In [822]:
df_train['Married']=le.fit_transform(df_train['Married'])

In [823]:
df_train.head()

Gender  Married Dependents  ... Credit_History Property_Area  Loan_Status
0       1        1          1  ...            1.0         Rural            N
1       1        1          0  ...            1.0         Urban            Y
2       1        1          0  ...            1.0         Urban            Y
3       1        0          0  ...            1.0         Urban            Y
4       1        1          2  ...            1.0         Urban            Y

[5 rows x 12 columns]

In [824]:
df_train['Married']=df_train['Married'].astype(int)

# **Dependents Analysis**

In [825]:
print('The number of unique values in the Dependents column is:\n',df_train['Dependents'].value_counts())

The number of unique values in the Dependents column is:
 0     274
2      85
1      80
3+     41
Name: Dependents, dtype: int64


In [826]:
for i in range(0,df_train.shape[0],1):
  if df_train['Dependents'][i]=='3+':
    df_train['Dependents'][i]='3'

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [827]:
df_train['Dependents']=le.fit_transform(df_train['Dependents'])

In [828]:
df_train.head()

Gender  Married  Dependents  ... Credit_History Property_Area  Loan_Status
0       1        1           1  ...            1.0         Rural            N
1       1        1           0  ...            1.0         Urban            Y
2       1        1           0  ...            1.0         Urban            Y
3       1        0           0  ...            1.0         Urban            Y
4       1        1           2  ...            1.0         Urban            Y

[5 rows x 12 columns]

In [829]:
df_train['Married']=df_train['Married'].astype(int)

# **Education Analysis**

In [830]:
print('The number of unique values in the Education column is:\n',df_train['Education'].value_counts())

The number of unique values in the Education column is:
 Graduate        383
Not Graduate     97
Name: Education, dtype: int64


In [831]:
df_train['Education']=le.fit_transform(df_train['Education'])

In [832]:
df_train.head()

Gender  Married  Dependents  ...  Credit_History Property_Area  Loan_Status
0       1        1           1  ...             1.0         Rural            N
1       1        1           0  ...             1.0         Urban            Y
2       1        1           0  ...             1.0         Urban            Y
3       1        0           0  ...             1.0         Urban            Y
4       1        1           2  ...             1.0         Urban            Y

[5 rows x 12 columns]

In [833]:
df_train['Education']=df_train['Education'].astype(int)

# **Self_Employed Analysis**

In [834]:
print('The number of unique values in the Self_Employed column is:\n',df_train['Self_Employed'].value_counts())

The number of unique values in the Self_Employed column is:
 No     414
Yes     66
Name: Self_Employed, dtype: int64


In [835]:
df_train['Self_Employed']=le.fit_transform(df_train['Self_Employed'])

In [836]:
df_train.head()

Gender  Married  Dependents  ...  Credit_History  Property_Area  Loan_Status
0       1        1           1  ...             1.0          Rural            N
1       1        1           0  ...             1.0          Urban            Y
2       1        1           0  ...             1.0          Urban            Y
3       1        0           0  ...             1.0          Urban            Y
4       1        1           2  ...             1.0          Urban            Y

[5 rows x 12 columns]

In [837]:
df_train['Self_Employed']=df_train['Self_Employed'].astype(int)

# **Self_Employed Analysis**

In [838]:
print('The number of unique values in the Self_Employed column is:\n',df_train['Self_Employed'].value_counts())

The number of unique values in the Self_Employed column is:
 0    414
1     66
Name: Self_Employed, dtype: int64


In [839]:
df_train['Self_Employed']=le.fit_transform(df_train['Self_Employed'])

In [840]:
df_train.head()

Gender  Married  Dependents  ...  Credit_History  Property_Area  Loan_Status
0       1        1           1  ...             1.0          Rural            N
1       1        1           0  ...             1.0          Urban            Y
2       1        1           0  ...             1.0          Urban            Y
3       1        0           0  ...             1.0          Urban            Y
4       1        1           2  ...             1.0          Urban            Y

[5 rows x 12 columns]

In [841]:
df_train['Self_Employed']=df_train['Self_Employed'].astype(int)

# **Credit_History Analysis**

In [842]:
print('The number of unique values in the Credit_History column is:\n',df_train['Credit_History'].value_counts())

The number of unique values in the Credit_History column is:
 1.0    410
0.0     70
Name: Credit_History, dtype: int64


In [843]:
df_train['Credit_History']=le.fit_transform(df_train['Credit_History'])

In [844]:
df_train.head()

Gender  Married  Dependents  ...  Credit_History  Property_Area  Loan_Status
0       1        1           1  ...               1          Rural            N
1       1        1           0  ...               1          Urban            Y
2       1        1           0  ...               1          Urban            Y
3       1        0           0  ...               1          Urban            Y
4       1        1           2  ...               1          Urban            Y

[5 rows x 12 columns]

In [845]:
df_train['Credit_History']=df_train['Credit_History'].astype(int)

# **Property_Area Analysis**

In [846]:
print('The number of unique values in the Property_Area column is:\n',df_train['Property_Area'].value_counts())

The number of unique values in the Property_Area column is:
 Semiurban    191
Urban        150
Rural        139
Name: Property_Area, dtype: int64


In [847]:
df_train['Property_Area']=le.fit_transform(df_train['Property_Area'])

In [848]:
df_train.head()

Gender  Married  Dependents  ...  Credit_History  Property_Area  Loan_Status
0       1        1           1  ...               1              0            N
1       1        1           0  ...               1              2            Y
2       1        1           0  ...               1              2            Y
3       1        0           0  ...               1              2            Y
4       1        1           2  ...               1              2            Y

[5 rows x 12 columns]

In [849]:
df_train['Property_Area']=df_train['Property_Area'].astype(int)

# **Loan_Status Analysis**

In [850]:
print('The number of unique values in the Loan_Status column is:\n',df_train['Loan_Status'].value_counts())

The number of unique values in the Loan_Status column is:
 Y    332
N    148
Name: Loan_Status, dtype: int64


In [851]:
df_train['Loan_Status']=le.fit_transform(df_train['Loan_Status'])

In [852]:
df_train.head()

Gender  Married  Dependents  ...  Credit_History  Property_Area  Loan_Status
0       1        1           1  ...               1              0            0
1       1        1           0  ...               1              2            1
2       1        1           0  ...               1              2            1
3       1        0           0  ...               1              2            1
4       1        1           2  ...               1              2            1

[5 rows x 12 columns]

In [853]:
df_train['Loan_Status']=df_train['Loan_Status'].astype(int)

# **Test Dataset**

# **Data Cleaning**

In [854]:
df_test=df_test.drop(columns=['Loan_ID'])
df_test

Gender Married Dependents  ... Loan_Amount_Term Credit_History  Property_Area
0     Male     Yes          0  ...            360.0            1.0          Urban
1     Male     Yes          1  ...            360.0            1.0          Urban
2     Male     Yes          2  ...            360.0            1.0          Urban
3     Male     Yes          2  ...            360.0            NaN          Urban
4     Male      No          0  ...            360.0            1.0          Urban
..     ...     ...        ...  ...              ...            ...            ...
362   Male     Yes         3+  ...            360.0            1.0          Urban
363   Male     Yes          0  ...            360.0            1.0          Urban
364   Male      No          0  ...            360.0            NaN      Semiurban
365   Male     Yes          0  ...            360.0            1.0          Rural
366   Male      No          0  ...            180.0            1.0          Rural

[367 rows x 11 columns]

In [855]:
df_test.shape

(367, 11)

In [856]:
df_test.describe()

ApplicantIncome  CoapplicantIncome  ...  Loan_Amount_Term  Credit_History
count       367.000000         367.000000  ...        361.000000      338.000000
mean       4805.599455        1569.577657  ...        342.537396        0.825444
std        4910.685399        2334.232099  ...         65.156643        0.380150
min           0.000000           0.000000  ...          6.000000        0.000000
25%        2864.000000           0.000000  ...        360.000000        1.000000
50%        3786.000000        1025.000000  ...        360.000000        1.000000
75%        5060.000000        2430.500000  ...        360.000000        1.000000
max       72529.000000       24000.000000  ...        480.000000        1.000000

[8 rows x 5 columns]

In [857]:
df_test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Gender             356 non-null    object 
 1   Married            367 non-null    object 
 2   Dependents         357 non-null    object 
 3   Education          367 non-null    object 
 4   Self_Employed      344 non-null    object 
 5   ApplicantIncome    367 non-null    int64  
 6   CoapplicantIncome  367 non-null    int64  
 7   LoanAmount         362 non-null    float64
 8   Loan_Amount_Term   361 non-null    float64
 9   Credit_History     338 non-null    float64
 10  Property_Area      367 non-null    object 
dtypes: float64(3), int64(2), object(6)
memory usage: 31.7+ KB


In [858]:
df_test['Married'].value_counts()

Yes    233
No     134
Name: Married, dtype: int64

In [859]:
#df_train=pd.get_dummies(df_train)

In [860]:
df_test.head()

Gender Married Dependents  ... Loan_Amount_Term Credit_History  Property_Area
0   Male     Yes          0  ...            360.0            1.0          Urban
1   Male     Yes          1  ...            360.0            1.0          Urban
2   Male     Yes          2  ...            360.0            1.0          Urban
3   Male     Yes          2  ...            360.0            NaN          Urban
4   Male      No          0  ...            360.0            1.0          Urban

[5 rows x 11 columns]

In [861]:
df_test=df_test.dropna()

In [862]:
df_test=df_test.reset_index()

In [863]:
df_test

index Gender Married  ... Loan_Amount_Term Credit_History Property_Area
0        0   Male     Yes  ...            360.0            1.0         Urban
1        1   Male     Yes  ...            360.0            1.0         Urban
2        2   Male     Yes  ...            360.0            1.0         Urban
3        4   Male      No  ...            360.0            1.0         Urban
4        5   Male     Yes  ...            360.0            1.0         Urban
..     ...    ...     ...  ...              ...            ...           ...
284    361   Male     Yes  ...            360.0            1.0     Semiurban
285    362   Male     Yes  ...            360.0            1.0         Urban
286    363   Male     Yes  ...            360.0            1.0         Urban
287    365   Male     Yes  ...            360.0            1.0         Rural
288    366   Male      No  ...            180.0            1.0         Rural

[289 rows x 12 columns]

In [864]:
df_test=df_test.iloc[:,1:]

In [865]:
df_test

Gender Married Dependents  ... Loan_Amount_Term Credit_History  Property_Area
0     Male     Yes          0  ...            360.0            1.0          Urban
1     Male     Yes          1  ...            360.0            1.0          Urban
2     Male     Yes          2  ...            360.0            1.0          Urban
3     Male      No          0  ...            360.0            1.0          Urban
4     Male     Yes          0  ...            360.0            1.0          Urban
..     ...     ...        ...  ...              ...            ...            ...
284   Male     Yes          1  ...            360.0            1.0      Semiurban
285   Male     Yes         3+  ...            360.0            1.0          Urban
286   Male     Yes          0  ...            360.0            1.0          Urban
287   Male     Yes          0  ...            360.0            1.0          Rural
288   Male      No          0  ...            180.0            1.0          Rural

[289 rows x 11 columns]

In [866]:
le=LabelEncoder()

# **Gender Analysis**

In [867]:
print('The number of unique values in the gender column is:\n',df_test['Gender'].value_counts())

The number of unique values in the gender column is:
 Male      230
Female     59
Name: Gender, dtype: int64


In [868]:
df_test['Gender']=le.fit_transform(df_test['Gender'])

In [869]:
df_test.head()

Gender Married Dependents  ... Loan_Amount_Term Credit_History  Property_Area
0       1     Yes          0  ...            360.0            1.0          Urban
1       1     Yes          1  ...            360.0            1.0          Urban
2       1     Yes          2  ...            360.0            1.0          Urban
3       1      No          0  ...            360.0            1.0          Urban
4       1     Yes          0  ...            360.0            1.0          Urban

[5 rows x 11 columns]

In [870]:
df_test['Gender']=df_test['Gender'].astype(int)

# **Married Analysis**

In [871]:
print('The number of unique values in the Married column is:\n',df_test['Gender'].value_counts())

The number of unique values in the Married column is:
 1    230
0     59
Name: Gender, dtype: int64


In [872]:
df_test['Married']=le.fit_transform(df_test['Married'])

In [873]:
df_test.head()

Gender  Married Dependents  ... Loan_Amount_Term Credit_History  Property_Area
0       1        1          0  ...            360.0            1.0          Urban
1       1        1          1  ...            360.0            1.0          Urban
2       1        1          2  ...            360.0            1.0          Urban
3       1        0          0  ...            360.0            1.0          Urban
4       1        1          0  ...            360.0            1.0          Urban

[5 rows x 11 columns]

In [874]:
df_test['Married']=df_test['Married'].astype(int)

# **Dependents Analysis**

In [875]:
print('The number of unique values in the Dependents column is:\n',df_test['Dependents'].value_counts())

The number of unique values in the Dependents column is:
 0     167
2      50
1      42
3+     30
Name: Dependents, dtype: int64


In [876]:
for i in range(0,df_test.shape[0],1):
  if df_test['Dependents'][i]=='3+':
    df_test['Dependents'][i]='3'

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [877]:
df_test['Dependents']=le.fit_transform(df_test['Dependents'])

In [878]:
df_test.head()

Gender  Married  Dependents  ... Loan_Amount_Term Credit_History  Property_Area
0       1        1           0  ...            360.0            1.0          Urban
1       1        1           1  ...            360.0            1.0          Urban
2       1        1           2  ...            360.0            1.0          Urban
3       1        0           0  ...            360.0            1.0          Urban
4       1        1           0  ...            360.0            1.0          Urban

[5 rows x 11 columns]

In [879]:
df_test['Married']=df_test['Married'].astype(int)

# **Education Analysis**

In [880]:
print('The number of unique values in the Education column is:\n',df_test['Education'].value_counts())

The number of unique values in the Education column is:
 Graduate        224
Not Graduate     65
Name: Education, dtype: int64


In [881]:
df_test['Education']=le.fit_transform(df_test['Education'])

In [882]:
df_test.head()

Gender  Married  Dependents  ...  Loan_Amount_Term Credit_History  Property_Area
0       1        1           0  ...             360.0            1.0          Urban
1       1        1           1  ...             360.0            1.0          Urban
2       1        1           2  ...             360.0            1.0          Urban
3       1        0           0  ...             360.0            1.0          Urban
4       1        1           0  ...             360.0            1.0          Urban

[5 rows x 11 columns]

In [883]:
df_test['Education']=df_test['Education'].astype(int)

# **Self_Employed Analysis**

In [884]:
print('The number of unique values in the Self_Employed column is:\n',df_test['Self_Employed'].value_counts())

The number of unique values in the Self_Employed column is:
 No     257
Yes     32
Name: Self_Employed, dtype: int64


In [885]:
df_test['Self_Employed']=le.fit_transform(df_test['Self_Employed'])

In [886]:
df_test.head()

Gender  Married  Dependents  ...  Loan_Amount_Term  Credit_History  Property_Area
0       1        1           0  ...             360.0             1.0          Urban
1       1        1           1  ...             360.0             1.0          Urban
2       1        1           2  ...             360.0             1.0          Urban
3       1        0           0  ...             360.0             1.0          Urban
4       1        1           0  ...             360.0             1.0          Urban

[5 rows x 11 columns]

In [887]:
df_test['Self_Employed']=df_test['Self_Employed'].astype(int)

# **Self_Employed Analysis**

In [888]:
print('The number of unique values in the Self_Employed column is:\n',df_test['Self_Employed'].value_counts())

The number of unique values in the Self_Employed column is:
 0    257
1     32
Name: Self_Employed, dtype: int64


In [889]:
df_test['Self_Employed']=le.fit_transform(df_test['Self_Employed'])

In [890]:
df_test.head()

Gender  Married  Dependents  ...  Loan_Amount_Term  Credit_History  Property_Area
0       1        1           0  ...             360.0             1.0          Urban
1       1        1           1  ...             360.0             1.0          Urban
2       1        1           2  ...             360.0             1.0          Urban
3       1        0           0  ...             360.0             1.0          Urban
4       1        1           0  ...             360.0             1.0          Urban

[5 rows x 11 columns]

In [891]:
df_test['Self_Employed']=df_test['Self_Employed'].astype(int)

# **Credit_History Analysis**

In [892]:
print('The number of unique values in the Credit_History column is:\n',df_test['Credit_History'].value_counts())

The number of unique values in the Credit_History column is:
 1.0    243
0.0     46
Name: Credit_History, dtype: int64


In [893]:
df_test['Credit_History']=le.fit_transform(df_test['Credit_History'])

In [894]:
df_test.head()

Gender  Married  Dependents  ...  Loan_Amount_Term  Credit_History  Property_Area
0       1        1           0  ...             360.0               1          Urban
1       1        1           1  ...             360.0               1          Urban
2       1        1           2  ...             360.0               1          Urban
3       1        0           0  ...             360.0               1          Urban
4       1        1           0  ...             360.0               1          Urban

[5 rows x 11 columns]

In [895]:
df_test['Credit_History']=df_test['Credit_History'].astype(int)

# **Property_Area Analysis**

In [896]:
print('The number of unique values in the Property_Area column is:\n',df_test['Property_Area'].value_counts())

The number of unique values in the Property_Area column is:
 Urban        113
Rural         89
Semiurban     87
Name: Property_Area, dtype: int64


In [897]:
df_test['Property_Area']=le.fit_transform(df_test['Property_Area'])

In [898]:
df_test.head()

Gender  Married  Dependents  ...  Loan_Amount_Term  Credit_History  Property_Area
0       1        1           0  ...             360.0               1              2
1       1        1           1  ...             360.0               1              2
2       1        1           2  ...             360.0               1              2
3       1        0           0  ...             360.0               1              2
4       1        1           0  ...             360.0               1              2

[5 rows x 11 columns]

In [899]:
df_test['Property_Area']=df_test['Property_Area'].astype(int)

# **Machine Learning Model**

In [900]:
from sklearn.linear_model import LinearRegression
from sklearn import metrics

In [901]:
X=df_train.iloc[:,0:11].values
y=df_train.iloc[:,11:].values

In [902]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.25,random_state=0)

In [903]:
lr=LinearRegression()
lr.fit(X_train,y_train)

LinearRegression()

In [904]:
pred=lr.predict(X_test)
#print('Accuracy',metrics.accuracy_score(y_test,pred))
print('r2 score',metrics.r2_score(y_test,pred))
print('mse',metrics.mean_squared_error(y_test,pred))


r2 score 0.22039427378489762
mse 0.1773603027139358


# **Logistic Regression**

In [905]:
from sklearn.linear_model import LogisticRegression
from sklearn import metrics

In [906]:
X=df_train.iloc[:,0:11].values
y=df_train.iloc[:,11:].values

In [907]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.25,random_state=0)

In [908]:
lr=LogisticRegression()
lr.fit(X_train,y_train)

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LogisticRegression()

In [909]:
pred=lr.predict(X_test)
print('Accuracy',metrics.accuracy_score(y_test,pred))
# print('r2 score',metrics.r2_score(y_test,pred))
# print('mse',metrics.mean_squared_error(y_test,pred))
print('Confution Matrix\n',metrics.confusion_matrix(y_test,pred))

Accuracy 0.775
Confution Matrix
 [[15 27]
 [ 0 78]]


# **XG Boost Classification**

In [910]:
from xgboost import XGBClassifier
from sklearn import metrics

In [911]:
X=df_train.iloc[:,0:11].values
y=df_train.iloc[:,11:].values

In [912]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.25,random_state=0)

In [913]:
lr=XGBClassifier()
lr.fit(X_train,y_train)

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_label.py:98: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_label.py:133: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


XGBClassifier()

In [914]:
pred=lr.predict(X_test)
print('Accuracy',metrics.accuracy_score(y_test,pred))
# print('r2 score',metrics.r2_score(y_test,pred))
# print('mse',metrics.mean_squared_error(y_test,pred))
print('Confution Matrix\n',metrics.confusion_matrix(y_test,pred))


Accuracy 0.7166666666666667
Confution Matrix
 [[13 29]
 [ 5 73]]


# **In this Loan Approval case Study the best suitable model is Logistic Regression of accuracy 0.775**